### ISSUE TITLE
using with search

### ISSUE THREAD

**Issue by dbinetti** [cibernox/ember-power-select-with-create#12]
Is it possible to use this with the 'search' api?  I poked around on the code and it *seems* like it  was designed to support that, but when I attempt to do so it never resolves the search (always 'loading options...')

Here is the markup:  
```
  {{#power-select-with-create
      search=(action "searchVenue")
      selected=venue
      placeholder='Venue'
      onchange=(action "saveRecord")
      oncreate=(action "createRecord")
      as |venue term|
}}
  {{venue.name}}
{{/power-select-with-create}}
```
and the controller code:

```
import Ember from 'ember';

export default Ember.Component.extend({
  store: Ember.inject.service(),
  actions: {
    createRecord(term) {
      console.log(term);
    },
    saveRecord(venue) {
      this.model.set('venue', venue);
      this.model.save();
    },
    searchVenue(term) {
      return new Ember.RSVP.Promise((resolve, reject) => {
        Ember.run.debounce(this, this._performSearch, term, resolve, reject, 600);
      });
    }
  },
  _performSearch(term, resolve, reject) {
    if (Ember.isBlank(term)) { return resolve([]); }
    this.get('store').query('venue', {'name__icontains': term})
      .then(data => resolve(data), reject);
  },
});
```

It is making a network request, which either returns a set (for matches) or an empty JSON API count (for no matches).  Thanks

**Comment by felixkiss** [comment]
Please see #18 for an example of using `power-select-with-create` with `search`

**Comment by thermokarst** [comment]
@dbinetti: Were you able to get this working? I don't see any real difference between your example and the test added in #18 (both components call an action that returns a promise. Maybe I am just missing something. Thanks!

**Comment by dbinetti** [comment]
No, i was not.  

I'm too new to Ember (and Javascript, for that matter) to write a proper test, but the reason I suspect this is a bug is as follows:

If I keep everything else the same and simply change `power-select-with-create`  back to `power-select`, it works.  I never get to the `oncreate` part because it never gets past "Loading options..."

**Comment by dbinetti** [comment]
And i guess i can't reopen this, but following #18 did not work.  Granted I am using Ember-Data and not fixtures so it might be there...

**Comment by felixkiss** [comment]
@dbinetti I have reopened this. This should be addressed by #23. Can you test and confirm it working for you?

**Comment by dbinetti** [comment]
worked for me
